SimulationAgent/DeviceReplay/DeviceReplayActor.cs

```diff
@@ -0,0 +1,212 @@
+            this.instance = instance;
+
+            this.status = ActorStatus.None;
+            this.deviceModel = null;
+            this.currentLine = "";
+            this.whenToRun = 0;
+            this.prevInterval = 0;
+            this.emptySchema = new DeviceModelMessageSchema();
+        }
+
+        /// <summary>
+        /// TODO: Fill in comment
+        /// </summary>
+        public void Init(
+            ISimulationContext simulationContext, 
+            string deviceId, 
+            DeviceModel deviceModel,
+            IDeviceConnectionActor context)
+        {
+            this.instance.InitOnce();
+
+            this.simulationContext = simulationContext;
+            this.deviceModel = deviceModel;
+            this.deviceId = deviceId;
+            this.deviceContext = context;
+            this.actorLogger.Init(deviceId, "Replay");
+
+            string path = deviceModel.ReplayFile;
+            try
+            {
+                // TODO: Pull from data store
+                if (File.Exists(path))
+                {
+                    this.fileStream = new FileStream(path, FileMode.Open, FileAccess.Read);
+                    this.streamReader = new StreamReader(this.fileStream);
+                    this.status = ActorStatus.ReadLine;
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("The process failed: {0}", e.ToString());
+            }
+
+            this.instance.InitComplete();
+        }
+
+        public bool HasWorkToDo()
+        {
+            if (Now < this.whenToRun) return false;
+
+            if (!this.deviceContext.Connected) return false;
+
+            switch (this.status) {
+                case ActorStatus.ReadLine:
+                case ActorStatus.LineReady:
+                case ActorStatus.Restart:
+                    return true;
+            }
+            return false;
+        }
+
+        public async Task RunAsync()
+        {
+            switch (this.status)
+            {
+                case ActorStatus.ReadLine:
+                    this.ReadLine();
+                    break;
+                case ActorStatus.LineReady:
+                    this.SendTelemetry();
+                    break;
+                case ActorStatus.Restart:
+                    // Rewind the stream to the beginning
+                    this.fileStream.Position = 0;
+                    this.streamReader.DiscardBufferedData();
+                    this.status = ActorStatus.ReadLine;
+                    break;
+            }
+        }
+
+        public void Stop()
+        {
+            this.log.Debug("Device replay actor stopped",
+                () => new { this.deviceId, Status = this.status.ToString() });
+
+            // Discard file reader resources
+            this.fileStream.Dispose();
+            this.streamReader.Dispose();
+
+            this.status = ActorStatus.Stopped;
+
+            // TODO: Stop the sim in the db
```

Comment:
We don't need this, correct ? ie we don't need to handle stopping sim from here.

Reply:
I thought so but I supposed not.